### PR TITLE
Update modules

### LINKED ISSUE #35: NF-CORE-MODULE: TCOFFEE_REGRESSIVE

### PR DESCRIPTION
**Pull request by luisas**
Closes #35, #45, #58, #60, #61

Add new modules (learnMSA and tcoffee align) from nf-core
Add local mtmalign module
Fix issues to enable structural alignments modes

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/multiplesequencealign/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, also make a PR on the nf-core/multiplesequencealign _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/compute_trees.nf [hunk 1]
```diff
@@ -20,17 +20,17 @@ include { CLUSTALO_GUIDETREE } from '../../modules/nf-core/clustalo/guidetree/ma
    ch_fastas_fortrees = ch_fastas.combine(tree_tools)
                                  .map( it -> [it[0] + it[2], it[1]] )
                                  .branch{
-                                          famsa_guidetree: it[0]["tree"] ==  "FAMSA_GUIDETREE"
-                                          mbed: it[0]["tree"] == "MBED"
+                                          famsa:    it[0]["tree"] == "FAMSA"
+                                          clustalo: it[0]["tree"] == "CLUSTALO"
                                         }
```
Comment (JoseEspinosa) [inline subworkflows/local/compute_trees.nf:25]:
```suggestion
  ch_fastas
      .combine(tree_tools)
      .map{ it -> [ it[0] + it[2], it[1] ] }
      .branch {
          famsa_guidetree: it[0]["tree"] ==  "FAMSA_GUIDETREE"
          mbed: it[0]["tree"] == "MBED"
          famsa:    it[0]["tree"] == "FAMSA"
          clustalo: it[0]["tree"] == "CLUSTALO"
      }
      .set { ch_fastas_fortrees }
```

File: workflows/multiplesequencealign.nf [hunk 2]
```diff
@@ -77,17 +78,27 @@ workflow MULTIPLESEQUENCEALIGN {
...
+                        alignMap["args_aligner_clean"] = WorkflowMultiplesequencealign.cleanArgs(meta_clone.args_aligner)
                         
                         [ treeMap, alignMap ]
                     }
 
 
     ch_seqs       = ch_input.map{ sample -> [ sample[0], file(sample[1]) ]}
     ch_refs       = ch_input.map{ sample -> [ sample[0], file(sample[2]) ]}
-    ch_structures = ch_input.map{ sample -> [ sample[0], sample[3]       ]}
+    ch_structures = ch_input.map{ sample -> [ sample[0], sample[3]       ]}.filter{ it[1].size() > 0 }
```
Comment (JoseEspinosa) [inline workflows/multiplesequencealign.nf:91]: This suggestions are to try to make the code as readable as possible
```suggestion
    ch_input
        .map { 
            sample -> [ sample[0], sample[3] ] 
        }
        .filter { it[1].size() > 0 }
        .set { ch_structures }
```

File: workflows/multiplesequencealign.nf [hunk 3]
```diff
@@ -77,17 +78,27 @@ workflow MULTIPLESEQUENCEALIGN {
...
+    
+    ch_structures.branch {
+        compressed:   it[1].endsWith('.tar.gz')
+        uncompressed: true
+    }
+    .set { ch_structures }
+
+    ch_structures = UNTAR ( ch_structures.compressed ).untar.mix( ch_structures.uncompressed )
```
Comment (JoseEspinosa) [inline workflows/multiplesequencealign.nf:99]:
```suggestion
    UNTAR ( ch_structures.compressed )
        .out
        .untar
        .mix( ch_structures.uncompressed )
        .set { ch_structures }
```

File: workflows/multiplesequencealign.nf [hunk 4]
```diff
@@ -77,17 +78,27 @@ workflow MULTIPLESEQUENCEALIGN {
...
+    ch_structures = ch_structures.map { meta,dir -> [meta,file(dir).listFiles().collect()]}
```
Comment (JoseEspinosa) [inline workflows/multiplesequencealign.nf:100]:
```suggestion
    ch_structures
        .map { 
            meta,dir -> 
                [ meta,file(dir).listFiles().collect() ] 
        }
        .set { ch_structures }
```

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
Just suggested some small changes, nice job!

File: modules/local/mtmalign_align.nf [hunk 5]
```diff
@@ -0,0 +1,25 @@
+process MTMALIGN_ALIGN {
+    tag "$meta.id"
```
[the remark this answers is not in the document]
Comment (luisas) [reply modules/local/mtmalign_align.nf:2]: There is none at the moment, if you prefer i can remove the module from the PR but maybe we can leave it and solve it in the next PR ( i already have an issue open for it and i plan to make it an nf-core module :)

File: assets/samplesheet_copy.csv [hunk 6]
```diff
@@ -0,0 +1,3 @@
+id,fasta,reference,structures
```
[the remark this answers is not in the document]
Comment (luisas) [reply assets/samplesheet_copy.csv:1]: This is just an extra file I am keeping now for testing - will remove it for the release :)

File: subworkflows/local/compute_trees.nf (hunk 1 above)
Comment (JoseEspinosa) [reply subworkflows/local/compute_trees.nf:25]: Yeah sorry, I copy paste it wrongly, I was complaining about the formatting of the code

Comment (JoseEspinosa) [reply subworkflows/local/compute_trees.nf:25]: I updated the comment but double check I did it correctly

Comment (luisas) [reply subworkflows/local/compute_trees.nf]: It would not allow me to change it here directly. I changed it directly in the code and pushed again.

**Review by JoseEspinosa — APPROVED** [review]
Awesome! 🚀

**Pull request merged**